Automattic/wp-calypso
# Plans Grid: show expired plan as spotlight

Fixes #85310

## Proposed Changes

* The free plan was shown as the current plan instead of the expired plan because the `getSitePlanSlug` selector returned the free plan if the current plan had expired. We were also using two different selectors (`getCurrentPlan` and `getSitePlanSlug`) which essentially did the same thing. In this PR, I have replaced the `getSitePlanSlug` selector with `currentPlan?.productSlug`, and it reduces the chance of a data mismatch.
* Added a new selector, `isCurrentPlanExpired`, which is similar to the existing `isCurrentPlanExpiring`. This is used to modify the text of the CTA on the spotlight plan to "Renew Plan" if the plan has expired.

### Screenshots
| Production | This PR |
|--------|--------|
|<img width="1903" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/521bf59d-31ef-4c24-9fb1-75ab994c133f"> |<img width="1906" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/a9945e09-8d6d-4f67-8ab0-d33e0d47a1ac"> |

## Testing Instructions



* Get a site with an expired plan (i.e. a few days past its expiration date). Automatticians can set this up themselves(using SA).
* Visit `/plans<site slug>` for your site, while logged in as the site owner.
* Confirm that the expired plan is shown in the spotlight card.
* Confirm that the CTA text is "Renew Plan".

## Pre-merge Checklist
